stdlib/src/builtin/string.mojo

```diff
@@ -1208,43 +1208,46 @@ struct String(
         res.append(0)
         return String(res^)
 
-    fn strip(self) -> String:
-        """Return a copy of the string with leading and trailing whitespace characters removed.
+    fn strip(self, chars: String = " \n\t\r\f\v") -> String:
```

Comment:
**suggestion** since the list of characters for whitespaces is fixed, we could define somewhere `alias WHITESPACES = " \n\t\r\f\v"` and use it in the default values to ensure the behavior is the same.

Reply:
Yeah, good idea. I thought about it, but forgot somehow. I'll probably add everything from https://github.com/python/cpython/blob/0085c3ae8f067abd4f6540d0f6dd2fb13107618e/Lib/string.py#L24C1-L32C62 now and will add additional tests

Reply:
Done.

Reply:
Maybe you can add other the constants in a separate pull request? I believe it will be more controversial than just `whitespaces`. Especially since there are functions already in the codebase that check for lowercase, uppercase, etc and those functions do not currently use those constants, they use a more optimized algorithm.

We wouldn't want to block the merge of strip, rstrip and lstrip because of a discussion concerning those new constants.

Reply:
I don't really know how to open different PR from same branch on GitHub.

About functions - python already have functions and constants too

Reply:
Btw, current implementation is using `String.WHITESPACE`

Reply:
You could make two branches if you want. https://github.com/modularml/mojo/pull/2555#discussion_r1591073450 , https://github.com/modularml/mojo/pull/2555#discussion_r1591067735 show that those new constants are more controversial than the strip function improvements. It could have value to separate those two changes (one PR for String.whitespace + strip,rstrip,lstrip and one PR for all the new constants) otherwise everything will be blocked until all discussions have been resolved. (it's just a tip from an old OSS dude to get your PRs merged faster, you're not obligated to follow it at all)